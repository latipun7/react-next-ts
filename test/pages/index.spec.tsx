import { render } from 'test/test.utils';
import Index from 'pages/index';

describe('Index pages', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Index />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('prints hello world!', () => {
    const { getByText } = render(<Index />);
    const element = getByText(/hello/i);

    expect(element).toHaveTextContent('Hello World!');
  });
});
