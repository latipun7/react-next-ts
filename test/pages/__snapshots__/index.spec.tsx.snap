// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index pages matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-k3wmp9"
  >
    <div
      class="css-h74a88"
    >
      <h2
        class="chakra-heading css-66ymwu"
      >
        with-chakra-ui-typescript
      </h2>
    </div>
    <div
      class="chakra-stack css-mhfnbu"
    >
      <p
        class="chakra-text css-0"
      >
        Hello World! Example repository of 
        <code
          class="chakra-code css-1baulvz"
        >
          Next.js
        </code>
         + 
        <code
          class="chakra-code css-1baulvz"
        >
          chakra-ui
        </code>
         + 
        <code
          class="chakra-code css-1baulvz"
        >
          typescript
        </code>
        .
      </p>
      <ul
        class="css-1nzymsn"
        role="list"
      >
        <li
          class="css-0"
        >
          <svg
            class="chakra-icon css-qld8i7"
            fill="currentColor"
            focusable="false"
            height="1em"
            role="presentation"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            class="chakra-link css-pxaocv"
            href="https://chakra-ui.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Chakra UI 
            <svg
              class="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.586 4.586a2 2 0 112.828 2.828l-3 3a2 2 0 01-2.828 0 1 1 0 00-1.414 1.414 4 4 0 005.656 0l3-3a4 4 0 00-5.656-5.656l-1.5 1.5a1 1 0 101.414 1.414l1.5-1.5zm-5 5a2 2 0 012.828 0 1 1 0 101.414-1.414 4 4 0 00-5.656 0l-3 3a4 4 0 105.656 5.656l1.5-1.5a1 1 0 10-1.414-1.414l-1.5 1.5a2 2 0 11-2.828-2.828l3-3z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </li>
        <li
          class="css-0"
        >
          <svg
            class="chakra-icon css-qld8i7"
            fill="currentColor"
            focusable="false"
            height="1em"
            role="presentation"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            class="chakra-link css-pxaocv"
            href="https://nextjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Next.js 
            <svg
              class="chakra-icon css-13otjrl"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M12.586 4.586a2 2 0 112.828 2.828l-3 3a2 2 0 01-2.828 0 1 1 0 00-1.414 1.414 4 4 0 005.656 0l3-3a4 4 0 00-5.656-5.656l-1.5 1.5a1 1 0 101.414 1.414l1.5-1.5zm-5 5a2 2 0 012.828 0 1 1 0 101.414-1.414 4 4 0 00-5.656 0l-3 3a4 4 0 105.656 5.656l1.5-1.5a1 1 0 10-1.414-1.414l-1.5 1.5a2 2 0 11-2.828-2.828l3-3z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
    <label
      class="chakra-switch css-xsxgo8"
    >
      <input
        class="chakra-switch__input"
        style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        type="checkbox"
        value=""
      />
      <span
        aria-hidden="true"
        class="chakra-switch__track css-14qxnv8"
      >
        <span
          class="chakra-switch__thumb css-0"
        />
      </span>
    </label>
    <footer
      class="css-1al6mwe"
    >
      <p
        class="chakra-text css-0"
      >
        Next ❤️ Chakra
      </p>
    </footer>
    <div
      class="css-1xr1j55"
    >
      <a
        class="chakra-link css-1heizff"
        href="https://chakra-ui.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="chakra-button css-bsfr8y"
          type="button"
        >
          chakra-ui
        </button>
      </a>
      <a
        class="chakra-link css-b5u8cn"
        href="https://github.com/vercel/next.js/blob/canary/examples/with-chakra-ui-typescript"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="chakra-button css-bsfr8y"
          type="button"
        >
          View Repo
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
